secure_uninstall/models/module.py

```diff
@@ -0,0 +1,72 @@
+# coding: utf-8
+# Copyright 2014 David BEAL @ Akretion <david.beal@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp.osv import orm
+from openerp import models, fields, api
+from openerp.exceptions import Warning as UserError
+from openerp.tools.config import config
+
+
+class Module(orm.Model):
+    _inherit = 'ir.module.module'
+
+    def button_uninstall(self, cr, uid, ids, context=None):
+        # can't be convert in new api because use button_uninstall fct
+        # in _button_immediate_function() is buggy in new api.
+        # Next v10 will solve it.
+        if not context:
+            context = {}
+        if 'uninstall_authorized' in context:
+            ctx = context.copy()
+            del ctx['uninstall_authorized']
```

Comment:
Because if I let this key, I obtain:
`maximum recursion depth exceeded while calling a Python object` traceback